javav2/example_code/cloudfront/Readme.md

```diff
@@ -4,67 +4,173 @@ This README discusses how to run and test the Java code examples for Amazon Clou
+| CreateCannedPolicyRequest   | `privateKeyFullPath`               | `privateKeyFullPath` is the full path to private key whose public key is used by the CreatePublicKey file.                                                                                                                                                                                                                                        |
+| CreateCustomPolicyRequest   | `privateKeyFullPath`               |                                                                                                                                                                                                                                                                                                                                                   |
+| SigningUtilities            |                                    | Contains methods that sign URLs and cookies.                                                                                                                                                                                                                                                                                                      |
+| DeleteDistribution          |                                    | Deletes the distribution.                                                                                                                                                                                                                                                                                                                         |
+| DeleteDistributionResources |                                    | Deletes the origin access control, key group, and public key.                                                                                                                                                                                                                                                                                     |
+
+`CloudFrontSigningTest` programmatically creates a CloudFront distribution along with the required setup 
+to test that signed URLs/cookies are required to access S3 items through CloudFront. As part of the setup, 
+the `CloudFrontSigningTest` prepares and deletes an S3 bucket along with uploading and deleting 
+the provided file, `index.html`. All resources created during the run of the tests are deleted at the end.
+
+The `CloudFrontSigningTest` tests take several minutes to run. Deploying the distribution, so that the test can run for signing URLs and cookies,
+takes a couple of minutes. Undeploying the distribution, so that it can be deleted, also takes a couple of minutes.
```

Comment:
The `CloudFrontSigningTest` deploys and undeploys the distribution, which takes several minutes. Deploying lets the test run for signing URLs and cookies, and undeploying lets the distribution get deleted.